src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,559 @@
+ * This array has information for each data point w.r.t shape, colors and on click callback along with
+ * x and y co-ordinates.
+ *
+ * @private
+ * @param {object} target - Object containing the subsets
+ * @returns {Array} List of data point subsets
+ */
+const getDataPointValues = (target) => target.internalValuesSubset;
+/**
+ * Checks the data-set is currently shown in the graph and if the y data-point value is null
+ * If they are then true, false otherwise
+ *
+ * @private
+ * @param {object} shownTargets - graph targets config object
+ * @param {object} value - data point value object
+ * @returns {boolean} true if data point needs to be hidden, false otherwise
+ */
+const shouldHideDataPoints = (shownTargets, value) =>
+    shownTargets.indexOf(value.key) < 0 || value.y === null;
+
+/**
+ * Gets all the circles given in the bubble graph.
+ *
+ * @private
+ * @returns {Array} List of circle nodes in bubble chart
+ */
+const getAllCircleNodes = () => {
+    const circleNodes = [];
+    const bubbleGroup = d3.selectAll(`.${styles.bubbleGraphContent}`);
+    bubbleGroup.each(function() {
+        const node = d3.select(this);
+        const bubbleNode = node.selectAll(`.${styles.point} circle`);
+        bubbleNode.each(function() {
+            const circle = d3.select(this);
+            circleNodes.push(...circle[0]);
+        });
+    });
+    return circleNodes;
+};
+
+/**
+ * Enforces blur state for all the bubbles that is not the one hovered on.
+ * This is provided regardless of whether onClick is present or not.
+ *
+ * @private
+ * @param {Array} nodes - List of d3.selection objects that are circles form bubble graph
+ * @returns {undefined} - returns nothing
+ */
+const enforceBubbleBlur = (nodes) =>
+    nodes.forEach((ele) => {
+        const node = d3.select(ele);
+        if (node.attr("aria-selected") === "false") {
+            node.classed(styles.bubbleBlur, true);
+        }
+    });
+
+/**
+ * Removes the carbon-bubbleBlur style from all the bubbles to unblur all the bubbles in the bubble graph.
+ *
+ * @private
+ * @param {Array} nodes - List of d3.selection objects that are circles in the bubbles graph.
+ * @returns {undefined} - returns nothing
+ */
+const removeBubbleBlur = (nodes) =>
+    nodes.forEach((ele) => {
+        const node = d3.select(ele);
+        if (node.attr("aria-selected") === "false") {
+            node.classed(styles.bubbleBlur, false);
+        } else {
+            node.attr("aria-selected", false);
+        }
+    });
+
+/**
+ * Handler for the bubble that is hovered on. It blurs all other bubble in the bubble graph except one which is selected.
+ *
+ * @private
+ * @param {HTMLElement} target - Target element bubble hovered on
+ * @param {string} hoverState - Mouse over or Mouse out
+ * @returns {undefined} - returns nothing
+ */
+const blurActionHandler = (target, hoverState) => {
+    d3.select(target.firstChild).attr("aria-selected", true);
+    const allCircleNodes = getAllCircleNodes();
```

Comment:
Instead of doing this, can we just blur all of them and just unblur the selected one? This will save all the preprocessing in `getAllCircleNodes`.

Reply:
I think we can't do that because we have to go inside each bubble to add the class blur,  the bubble which is selected we don't have to add any class to that bubble?

Reply:
It's the same amount of processing you are doing now. Currently you are adding a class called blur to all the elements and no adding to the selected one. 
What I am suggesting is doing the same for all elements and un-blur the selected one.

Reply:
> since the properties like "fill-opacity and stroke-opacity" are inside the circle

You are just adding another class along with `carbon-point`. Also you dont need to add a separate class called `carbon-bubble-blur` you can just add a CSS for aria-selected attribute. Something like this: https://github.com/cerner/carbon-graphs/blob/master/src/main/less/core.less#L169

Reply:
I did the change did that look good.